src/classes/VOL_CTRL_VolunteersJobListingFS_TEST.cls

```diff
@@ -30,8 +30,109 @@
 
 @isTest
 private with sharing class VOL_CTRL_VolunteersJobListingFS_TEST {
+    private static VOL_CTRL_VolunteersJobListingFS jobListingCtrl;
 
     //==================== TEST METHOD(s) ======================================
+    @IsTest
+    private static void shouldReturnOneShiftWhenJobIdAndShiftIdAreProvidedWithZeroMonths() {
+        generateData(10);
+        Id jobId = [SELECT Id FROM Volunteer_Job__c LIMIT 1].Id;
+        Id shiftId = [SELECT Id FROM Volunteer_Shift__c Where Volunteer_Job__c = :jobId LIMIT 1].Id;
+        setupPage(new Map<String, String>{'jobId' => jobId, 'volunteerShiftId' => shiftId, 'nMonthsToShow' => '0'});
+
+        System.assertEquals(shiftId, jobListingCtrl.shiftIdFilter);
+        System.assertEquals(1, jobListingCtrl.listVolunteerJobs[0].Volunteer_Job_Slots__r.size(), 'The number of shifts returned was not as expected.');
+    }
+
+    @IsTest
+    private static void shouldReturnAllShiftsWhenJobIdIsProvidedAndLessThanLimit() {
+        generateData(10);
+        Id jobId = [SELECT Id FROM Volunteer_Job__c LIMIT 1].Id;
+        setupPage(new Map<String, String>{'jobId' => jobId});
+
+        System.assertEquals(jobId, jobListingCtrl.jobIdFilter);
+        System.assertEquals(10, jobListingCtrl.listVolunteerJobs[0].Volunteer_Job_Slots__r.size(), 'The number of shifts returned was not as expected.');
+    }
+
+    @IsTest
+    private static void shouldReturnAllShiftsWhenCampaingIdIsProvidedAndLessThanLimit() {
+        generateData(10);
+        Id campaignId = [SELECT Id FROM Campaign LIMIT 1].Id;
+        setupPage(new Map<String, String>{'campaignId' => campaignId});
+
+        System.assertEquals(campaignId, jobListingCtrl.campaignIdFilter);
+        System.assertEquals(10, jobListingCtrl.listVolunteerJobs[0].Volunteer_Job_Slots__r.size(), 'The number of shifts returned was not as expected.');
+    }
+
+    @IsTest
+    private static void shouldRetrunLimitedShiftsWhenIdFiltersAreNotProvided() {
```

Comment:
"Return" typo here and copy/pasted in a few methods below.

Reply:
Thanks @scottwarren-sfdo for catching these!